File: tests/unit/src/testFixtures/java/io/buildfoundation/bazel/detekt/execute/TestExecutable.java
package io.buildfoundation.bazel.detekt.execute;

public class TestExecutable implements Executable {

    private final ExecutableResult executeResult;

    public TestExecutable(ExecutableResult executeResult) {
        this.executeResult = executeResult;
    }

    @Override
    public ExecutableResult execute(String[] args) {
        return executeResult;
    }
}

File: detekt/wrapper/src/testFixtures/kotlin/io/buildfoundation/bazel/detekt/TestExecutable.kt
package io.buildfoundation.bazel.detekt

import io.buildfoundation.bazel.detekt.Executable
import io.buildfoundation.bazel.detekt.Result

internal class TestExecutable(private val result: Result) : Executable {

    override fun execute(args: Array<String>) = result
}